src/components/stack/BlockItem.js

```diff
@@ -12,21 +12,180 @@ class BlockItem extends Component {
 
     this.state = {
       hover: false,
-      modTitle: null,
-      modDescription: null,
-      modDurationWork: null,
-      modDurationBreak: null,
+      modTitle: null, //title of the task block
+      modDescription: null, //description of the task block
+      modDurationWork: null, //time length in seconds of burst
+      modDurationBreak: null, //time length in seconds of break
       focusState: FOCUS_NONE,
     };
+
+    this.handleOnHover = this.handleOnHover.bind(this);
+    this.handleOnNotHover = this.handleOnNotHover.bind(this);
+    this.handleOnClick = this.handleOnClick.bind(this);
+    this.handleCancel = this.handleCancel.bind(this);
+    this.handleEdit = this.handleEdit.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+  }
+
+  //onMassEnter and onMassLeave
+  //onhover, check what state it's in, based on that change the local state
+
+  handleOnHover = () => {
+    document.body.style.cursor = 'grab';
+    //check what state we are in first, then determine what's next like
+    //a finite state machine
+    this.setState({ focusState: FOCUS_SHOW_BUTTONS });
+  }
+
+  handleOnNotHover = () => {
+    console.log("MOUSE EXITED");
+    this.setState({ focusState: FOCUS_NONE });
+    //document.body.style.cursor = ''; //default arrow cursor
+  }
+
+  handleOnClick = () => {
+    this.setState({ focusState: FOCUS_SHOW_INFO });
+  }
+
+  handleCancel = () => {
+    this.setState({ focusState: FOCUS_SHOW_INFO });
+  }
+
+  handleEdit = () => {
+    this.setState({ focusState: FOCUS_EDIT });
+  }
+
+  handleClose = () => {
+    this.setState({ focusState: FOCUS_NONE });
   }
 
   render() {
     const { blocks, blockId } = this.props;
+    let jsxRendering;
```

Comment:
Please use a more descriptive name. Doesn't have to be too fancy since I know it'll these forms can be abstracted to another file.

Reply:
Will do. I was having trouble coming up with a name for this lol but I'll come up with a clearer one.